Blockzilla/URLBar.swift

```diff
@@ -662,17 +664,21 @@ class URLBar: UIView {
     }
 
     fileprivate func setTextToURL() {
-        var fullUrl: String? = nil
+        var displayText: String? = nil
         var truncatedURL: String? = nil
 
         if let url = url {
             // Strip the username/password to prevent domain spoofing.
             var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
             components?.user = nil
             components?.password = nil
-            fullUrl = components?.url?.absoluteString
+            
+            if let searchedText = SearchHistoryUtils.pullSearchFromStack(), searchedText != "" {
+                displayText = searchedText
+            }
+            
             truncatedURL = components?.host
-            urlText.text = isEditing ? fullUrl : truncatedURL
```

Comment:
This reverts one of the 7.0 PRs that displays a truncated version of the URL except when editing. We now need to have three states (search query if a search, full URL when editing, truncated url otherwise). All URLs should still display and act exactly the same as before, only search queries changing.